File: CommonCoreLegacy/src/main/java/com/skedgo/tripkit/common/model/MiniInstruction.java
package com.skedgo.tripkit.common.model;

import androidx.annotation.Nullable;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import org.immutables.gson.Gson;
import org.immutables.value.Value;


@Gson.TypeAdapters
@Value.Immutable
@Value.Style(passAnnotations = JsonAdapter.class)
@JsonAdapter(GsonAdaptersMiniInstruction.class)
public abstract class MiniInstruction {
    @SerializedName("description") @Nullable public abstract String getDescription();
    @SerializedName("instruction") @Nullable public abstract String getInstruction();
    @SerializedName("mainValue") @Nullable public abstract String getMainValue();
}

File: CommonCoreLegacy/src/main/java/com/skedgo/tripkit/common/model/MiniInstruction.kt
package com.skedgo.tripkit.common.model

import com.google.gson.annotations.JsonAdapter
import com.google.gson.annotations.SerializedName
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@TypeAdapters
@Immutable
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(
    GsonAdaptersMiniInstruction::class
)
abstract class MiniInstruction {
    @get:SerializedName("description")
    abstract val description: String?

    @get:SerializedName("instruction")
    abstract val instruction: String?

    @get:SerializedName("mainValue")
    abstract val mainValue: String?
}